dockstore-webservice/src/main/java/io/dockstore/webservice/helpers/AbstractImageRegistry.java

```diff
@@ -419,18 +425,91 @@ private void updateFiles(Tool tool, Tag tag, final FileDAO fileDAO, SourceCodeRe
             tag.addSourceFile(file);
         }
 
-        // need to go through all files for the booleans
-        boolean hasDockerfile = tag.getSourceFiles().stream().anyMatch(file -> file.getType() == SourceFile.FileType.DOCKERFILE);
+        // Update the tag with validation information
+        tag = validateTagDockerfile(tag, tool.isPrivateAccess());
+        tag = validateTagDescriptorType(tag, SourceFile.FileType.DOCKSTORE_CWL, tag.getCwlPath());
+        tag = validateTagDescriptorType(tag, SourceFile.FileType.DOCKSTORE_WDL, tag.getWdlPath());
+
+        boolean isValidVersion = isValidVersion(tag);
+        tag.setValid(isValidVersion);
+    }
+
+    /**
+     * Checks if the given tag is valid given its version validations.
+     * TODO: Duplicate in HostedToolResource.java
+     * @param tag Tag to check validation
+     * @return True if valid tag, false otherwise
+     */
+    private boolean isValidVersion(Tag tag) {
+        SortedSet<VersionValidation> versionValidations = tag.getValidations();
```

Comment:
Worth an experiment, will this stop people from being able to launch previously valid workflows?